src/Octoshift/GithubApi.cs

```diff
@@ -644,8 +644,18 @@ public virtual async Task<string> GetArchiveMigrationStatus(string org, int migr
 
             var response = await _client.GetAsync(url);
             var data = JObject.Parse(response);
+            var state = (string)data["state"];
 
-            return (string)data["state"];
+            if (state == ArchiveMigrationStatus.Failed)
+            {
+                var result = await _retryPolicy.RetryOnResult(async () => await GetArchiveMigrationStatus(org, migrationId), ArchiveMigrationStatus.Failed);
+
+                state = result.Outcome == Polly.OutcomeType.Failure
+                    ? throw new OctoshiftCliException($"Archive generation failed for id: {migrationId}")
+                    : result.Result;
+            }
```

Comment:
This won't work as expected and will cause an infinite loop probably because we're recursively calling the method! It would work in the handler but here since it's a lower lever we need to also retry the lower level calls like this:
```c#
public virtual async Task<string> GetArchiveMigrationStatus(string org, int migrationId)
{
    var url = $"{_apiUrl}/orgs/{org}/migrations/{migrationId}";

    var response = await _retryPolicy.RetryOnResult(async () =>
    {
        var httpResponse = await _client.GetAsync(url);
        var data = JObject.Parse(httpResponse);

        return (string)data["state"];
    }, ArchiveMigrationStatus.Failed);

    return response.Outcome == OutcomeType.Failure
        ? throw new OctoshiftCliException($"Archive generation failed for id: {migrationId}")
        : response.Result;
}
```

Reply:
Oh! Derp- good point! Committed these changes in: 431fffdc5085540df5df70e6eaab633d6d5f89fd